proposals/v3/api.md

```diff
@@ -0,0 +1,1119 @@
+is provided:
+
+```json
+{
+  "id": "56a32eba-1aa6-4868-84ee-fe01af8b2e6b",
+  "state": "ok",
+  "status": "done",
+  "message": "",
+  "components": [
+    "b9pin8ofmg5g01vmt77g"
+  ]
+}
+```
+
+**Fields**
+
+| Field | Description |
+| ----- | ----------- |
+| *id* | The primary ID of the transaction. |
+| *state* | The "least complete" state of the component transactions. (`ok`, `error`) |
+| *status* | The "least complete" status of the component transactions. (`unknown`, `pending`, `writing`, `done`) |
+| *message* | Any context information relating to a transaction's error state. If there is no error, this will be an empty string. |
+| *components* | The ID(s) for the component device write transaction(s). |
+
+Below is an example of the 200 OK response from the `/transaction` endpoint when a component device write transaction ID
+is provided:
+```json
+{
+  "id": "b9pin8ofmg5g01vmt77g",
+  "timeout": "10s",
+  "device": "0fe8f06229aa9a01ef6032d1ddaf18a5",
+  "context": {
+    "action": "color",
+    "data": "f38ac2"
+  },
+  "state": "ok",
+  "status": "done",
+  "created": "2018-02-01T15:00:51.132823149Z",
+  "updated": "2018-02-01T15:00:51.132823149Z",
+  "message": ""
+}
+```
+
+**Fields**
+
+| Field | Description |
+| ----- | ----------- |
+| *id* | The ID of the transaction. This is generated and assigned by the plugin being written to. |
+| *timeout* | A string representing the timeout for the write transaction after which it will be cancelled. This is effectively the maximum wait time for the transaction to resolve. |
+| *device* | The GUID of the device being written to. |
+| *context* | The POSTed write data for the given write transaction. |
+| *state* | The current state of the transaction. (`ok`, `error`) |
+| *status* | The current status of the transaction. (`unknown`, `pending`, `writing`, `done`) |
+| *created* | The time at which the transaction was created. This timestamp is generated by the plugin. |
+| *updated* | The last time the transaction state *or* status was updated. Once the transaction reaches a `done` status or `error` state, no further updates will occur. |
+| *message* | Any context information relating to a transaction's error state. If there is no error, this will be an empty string. |
+
+
+Below is an example of the 200 OK response from the `/transaction` endpoint when no transaction ID
+is provided:
+
+```json
+[
+  "56a32eba-1aa6-4868-84ee-fe01af8b2e6b",
+  "56a32eba-1aa6-4868-84ee-fe01af8b2e6c",
+  "56a32eba-1aa6-4868-84ee-fe01af8b2e6d"
+]
+```
+
+##### Error (500, 404)
+See: [Errors](#errors)
+
+#### States
+Transactions can be in one of two states:
+
+| State | Description |
+| ----- | ----------- |
+| *ok* | The default state. This signifies that the transaction is being processed regularly and has encountered no errors. |
+| *error* | The error state. This signifies an error with handling or resolving the transaction at any point during its lifecycle. |
+
+#### Status
```

Comment:
I can see how it could be confusing. We could simplify this to only have one, something like
```
pending --> writing --> done
    |          |
    +----------+------> error
```
where:
* `pending`: the transaction was received and created and is awaiting processing
* `writing`: the transaction is being processed
* `done`: (terminal state) the transaction finished processing successfully
* `error`: (terminal state) the transaction finished processing unsuccessfully

Reply:
updated transaction states per my previous comment (e4a4cb6)